packages/content-management/table_list_view_table/src/components/table.tsx

```diff
@@ -150,7 +152,17 @@ export function Table<T extends UserContentCommonSchema>({
     totalActiveFilters,
   } = useTagFilterPanel({
     query: searchQuery.query,
-    getTagList,
+    getTagList: useCallback(() => {
+      const tags = getTagList();
+      // In some situation, tags are created after page load.
+      // Therefore the getTagList result is cached before tags created
+      // Do this check to make sure default selected tags are showed
+      if (tags.length === 0 && tagReferences != null) {
+        return tagReferences;
+      }
+      return tags;
+    }, [getTagList, tagReferences]),
```

Comment:
If we don't do the check, it returns the value cached on page load, but there are some scenario that tags create after page loaded:
<img width="2555" alt="incorrect_cache_tag_count" src="https://github.com/elastic/kibana/assets/6295984/7aaab121-7acc-40e8-b8a0-78efd4f07c98">

After adding the check for initial tag references:
<img width="2545" alt="corrected_tag_count" src="https://github.com/elastic/kibana/assets/6295984/4ff9e3de-470d-4c2c-9c44-72438c05b1e6">